### PR TITLE
Clarification to JWT docs

### PR DESCRIPTION
**Pull request by ZacharyKlein**

### CODE REVIEW

File: src/main/docs/guide/security/authenticationStrategies/jwt/reader/bearerToken.adoc [hunk 1]
```diff
@@ -1,25 +1,21 @@
-Micronaut supports https://tools.ietf.org/html/rfc6750[RFC 6750 Bearer Token] specification out-of-the-box.
-
-It is enabled by default but requires JWT Authentication to be enabled.
```
Comment (sdelamo) [inline src/main/docs/guide/security/authenticationStrategies/jwt/reader/bearerToken.adoc]: You will need `micronaut.security.token.jwt.enabled=true`. Probably, we should not remove that sentence.

File: src/main/docs/guide/security/authenticationStrategies/jwt/reader/bearerToken.adoc [hunk 2]
```diff
@@ -1,25 +1,21 @@
...
-
-The following sequence illustrates the authentication flow:
+Micronaut supports the https://tools.ietf.org/html/rfc6750[RFC 6750 Bearer Token] specification for transmitting JWT tokens. The following sequence illustrates the RFC 6750 authentication flow:
 
 image::jwt-bearer-token.svg[]
 
-The following configuration properties are available to customize Bearer Token read:
+The following configuration properties are available to customize how the Bearer Token will be embedded in the request:
```
Comment (sdelamo) [inline src/main/docs/guide/security/authenticationStrategies/jwt/reader/bearerToken.adoc]: Actually, the configurations described are about how the token is read. No how the token is embedded in the request.

File: src/main/docs/guide/security/authenticationStrategies/jwt/reader/bearerToken.adoc [hunk 3]
```diff
@@ -1,25 +1,21 @@
...
 
-[source, bash]
-.Accessing a protected resource using Authorization request header
+[source, groovy]
+.Accessing a protected resource using an Authorization request header
 ----
-GET /protectedResource HTTP/1.1
-Host: micronaut.example
-Authorization: Bearer eyJhbGciOiJIUzI1NiJ9.eyJleHAiOjE0MjI5OTU5MjIsInN1YiI6ImppbWkiLCJyb2xlcyI6WyJST0xFX0FETUlOIiwiUk9MRV9VU0VSIl0sImlhdCI6MTQyMjk5MjMyMn0.rA7A2Gwt14LaYMpxNRtrCdO24RGrfHtZXY9fIjV8x8o
+String accessToken = //... obtain the access token
```
Comment (sdelamo) [inline src/main/docs/guide/security/authenticationStrategies/jwt/reader/bearerToken.adoc]: I will not remove the code which shows the request without micronaut code. If you want to add a code example. I will use the method: `HttpRequest.GET('/' ).bearer(accessToken)`

Comment (ZacharyKlein) [reply src/main/docs/guide/security/authenticationStrategies/jwt/reader/bearerToken.adoc]: My issue with the existing code snippet was that (in my opinion) it didn't do what the docs claimed: "[show] how to send a JWT token in the Authorization request header". The snippet is just a raw HTTP request with the auth header. It's not something one would ever write in a Micronaut app. 

If we're wanting to show what a request auth header looks like, that's fine with me, but I still think we should include a code example that uses Micronaut's client to make the request with the header and token.

File: src/main/docs/guide/security/authenticationStrategies/jwt/reader/bearerToken.adoc (hunk 2 above)
Comment (ZacharyKlein) [reply src/main/docs/guide/security/authenticationStrategies/jwt/reader/bearerToken.adoc]: Thanks, will revert that change.

File: src/main/docs/guide/security/authenticationStrategies/jwt/reader/bearerToken.adoc (hunk 1 above)
Comment (ZacharyKlein) [reply src/main/docs/guide/security/authenticationStrategies/jwt/reader/bearerToken.adoc]: As discussed in Slack, I think everything in this section of the docs (`15.3.3`) presumes that `micronaut.security.token.jwt.enabled=true`. So if we're repeating that setting here (`15.3.3.1`), I think for consistency we should do the same thing for `JWT Token Generation`(`15.3.3.2`) and `JWT Token Validation`(`15.3.3.3`), as they are sibling sub-sections.

**Pull request merged**